### PR TITLE
auth0: implement identity provider

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Auth0 supports OIDC out of the box, so here's a basic implementation of their identity provider so we can use the type `auth0` instead of just the generic one.

## Related issues
- #1453 

**Checklist**:
- [x] add related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

File: internal/identity/oidc/auth0/auth0.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Package auth0 implements OpenID Connect for auth0
+//
+// https://www.pomerium.io/docs/identity-providers/auth0.html
```
Comment (desimone) [inline internal/identity/oidc/auth0/auth0.go:3]: 👀

**Review by desimone — COMMENTED** [review]
@calebdoxsey LGTM; maybe we could collaborate with the Auth0 folks on some shared docs? 

\cc @grounded042 \ @lxfontes

File: internal/identity/oidc/auth0/auth0.go (hunk 1 above)
Comment (calebdoxsey) [reply internal/identity/oidc/auth0/auth0.go:3]: yeah definitely. We can do that in a subsequent PR once we know what will be needed for the service account.

Comment (calebdoxsey) [reply internal/identity/oidc/auth0/auth0.go:3]: @desimone I added a basic markdown guide which we can add to.

**Comment by grounded042** [comment]
This looks great @calebdoxsey! Thanks for putting this together so quickly. I can definitely take care of adding docs on this as part of implementing the `directory` component. I'm planning on carving out some time this week to do that work.

File: docs/docs/identity-providers/auth0.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+title: Okta
```
Comment (travisgroth) [inline docs/docs/identity-providers/auth0.md:2]:
```suggestion
title: Auth0
```

File: docs/docs/identity-providers/auth0.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+[Log in to your Auth0 account](https://manage.auth0.com/) and head to your dashboard. Select **Applications** on the left menu. On the Applications page, click the **Create Application** button to create a new app.
+
+![Auth0 Applications Dashboard](./img/auth0/dashboard.png)
+
+## Create Regular Web Application
+
+On the **Create New Application** page, select the **Regular Web Application** for your application.
+
+![Okta Create Application Select Platform](./img/auth0/create.png)
```
Comment (travisgroth) [inline docs/docs/identity-providers/auth0.md:20]:
```suggestion
![Auth0 Create Application Select Platform](./img/auth0/create.png)
```

**Review by desimone — APPROVED** [review]
Good stuff!

**Pull request merged**